services/statestorage/merkle/merkle_forest.go

```diff
@@ -0,0 +1,236 @@
+package merkle
+
+import (
+	"fmt"
+	"github.com/orbs-network/orbs-network-go/crypto/hash"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+type RootId uint64
+type Proof []*Node
+
+var zeroValueHash = hash.CalcSha256([]byte{})
+
+type Node struct {
+	path     string
+	value    primitives.Sha256
+	branches [256]primitives.MerkleSha256
```

Comment:
256 I would move to a const to make more readable where this "magic" number comes from